File: src/main/java/codingdojo/TaxCalculator.java
package codingdojo;

import java.text.ParseException;
import java.util.Collection;

import static codingdojo.Util.fromISO8601UTC;

public class TaxCalculator {

    public static double calculateAddedTax(Collection<Order> orders) throws ParseException {
        double tax = 0D;
        for (Order order : orders) {
            if (order.getDate().before(fromISO8601UTC("2018-01-01T00:00Z"))) {
                tax += 10;
            } else {
                tax += 20;
            }
            for (Product product : order.getProducts()) {
                if (product.isEvent()) {
                    tax += product.getPrice().getAmountInCurrency("USD")* 0.25;
                } else {
                    tax += product.getPrice().getAmountInCurrency("USD")* 0.175;
                }
            }
        }
        return tax;
    }
}

File: CSharp/ProductExport/TaxCalculator.cs
using System.Collections.Generic;

namespace ProductExport
{
    public class TaxCalculator
    {
        public static double CalculateAddedTax(List<Order> orders)
        {
            var tax = 0D;
            foreach (var order in orders)
            {
                if (order.Date < Util.FromIsoDate("2018-01-01T00:00Z"))
                    tax += 10;
                else
                    tax += 20;

                foreach (var product in order.Products)
                    if (product.IsEvent())
                        tax += product.Price.GetAmountInCurrency("USD") * 0.25;
                    else
                        tax += product.Price.GetAmountInCurrency("USD") * 0.175;
            }

            return tax;
        }
    }
}